src/Disks/ObjectStorages/Local/LocalObjectStorage.cpp

```diff
@@ -140,16 +168,10 @@ ObjectMetadata LocalObjectStorage::getObjectMetadata(const std::string & /* path
 void LocalObjectStorage::copyObject( // NOLINT
     const StoredObject & object_from, const StoredObject & object_to, std::optional<ObjectAttributes> /* object_to_attributes */)
 {
-    fs::path to = object_to.absolute_path;
-    fs::path from = object_from.absolute_path;
-
-    /// Same logic as in DiskLocal.
-    if (object_from.absolute_path.ends_with('/'))
-        from = from.parent_path();
-    if (fs::is_directory(from))
-        to /= from.filename();
-
-    fs::copy(from, to, fs::copy_options::recursive | fs::copy_options::overwrite_existing);
+    auto in = readObject(object_from);
+    auto out = writeObject(object_to, WriteMode::Rewrite);
+    copyData(*in, *out);
```

Comment:
Why we need to read/write? Why we cannot copy blob from `object_from` to `object_to`?

Reply:
because the point is to not use std::filesystem::copy (for https://github.com/ClickHouse/ClickHouse/issues/42337)